conda/shell/condabin/_conda_activate.bat

```diff
@@ -38,10 +38,10 @@
 )
 @ECHO Failed to create temp directory "%TMP%\conda-<RANDOM>\" & exit /b 1
 :tmp_file_created
-@"%CONDA_EXE%" %_CE_M% %_CE_CONDA% shell.cmd.exe %* 1>%UNIQUE%
+@"%CONDA_EXE%" %_CE_M% %_CE_CONDA% shell.cmd.exe %* 1>"%UNIQUE%"
 @IF %ErrorLevel% NEQ 0 @EXIT /B %ErrorLevel%
-@FOR /F %%i IN (%UNIQUE%) DO @SET _TEMP_SCRIPT_PATH=%%i
-@RMDIR /S /Q %UNIQUE_DIR%
+@SET /P _TEMP_SCRIPT_PATH=<"%UNIQUE%"
+@RMDIR /S /Q "%UNIQUE_DIR%"
```

Comment:
Hey @Hepson, could you elaborate what's going on here, why you changed the part where it iterates over the list of uniques?

Reply:
@jezdez looks like `%UNIQUE%` is the first valid `!UNIQUE_DIR!\conda.tmp` path (see line 30-37 above) so no need to iterate over it?

Reply:
@jezdez  As per my understanding, we store the location of a unique temporary script in the "UNIQUE" file (_Aka. `@"%CONDA_EXE%" %_CE_M% %_CE_CONDA% shell.cmd.exe %* 1>"%UNIQUE%"`_) that we try to retrieve.

The previous loop allows retrieving the last segment of this file content. However, the space seems to be a delimiter that will conclude in retrieving a partial path. Instead, I propose to replace it by reading the first line.

This solution will only work if we have effectively just one line, but I haven't the knowledge to assume it. I've forgotten the reason since. When I made these modifications, I was more confident that we get just one entry. I think that is a proper solution based on unit tests and functional tests.

Reply:
As per my understanding, `conda shell.cmd.exe activate` only returns a single line with a temporary path to the activation script:

```
$ conda shell.cmd.exe activate
/var/folders/1b/z374sygs1s96t_xxxv0hr3lr0000gn/T/tmpot5p01v9.bat                   
```

So this seems to be safe.

---

Full explanation

1. User calls `conda activate "my env"`
2. This enters the `conda` shell command, defined in `conda/shell/condabin/conda.bat`. Since we asked for `activate`, it forwards to `conda/shell/condabin/_conda_activate.bat`, where this PR is adding changes.
3. `_conda_activate.bat` [will call](https://github.com/conda/conda/blob/92cfafcd65c338240d562d7a74cf7285f2eb5670/conda/shell/condabin/_conda_activate.bat#L41) `conda shell.cmd.exe <activate|deactivate>`, which takes us to `conda.cli.main`. There, `source`'d commands are [split](https://github.com/conda/conda/blob/92cfafcd65c338240d562d7a74cf7285f2eb5670/conda/cli/main.py#L125) from `install`-like commands. 
4. `main_sourced` will delegate to the relevant activator class. Since we have `cmd.exe`, then it calls [`conda.activate.CmdExeActivator.execute()`](https://github.com/conda/conda/blob/92cfafcd65c338240d562d7a74cf7285f2eb5670/conda/activate.py#L169).
5. `.execute()` delegates to the relevant subcommand (`activate` in our case), which delegates to `_finalize()` (a method that writes a given string to stdout OR a temporary file and returns its path).
6. This temporary path is printed to STDOUT.
7. We are back to `_conda_activate.bat`, which will now `@CALL` that temporary path.

As a result, we can see that `_finalize()` will only return the temporary path on Windows, so it's a single line, so this method is safe.